### PR TITLE
Add mention on question mark escape feature

### PR DESCRIPTION
**Pull request by julienfastre**
Since PHP 7.4, it is possible to escape question mark in php pdo statement by doubling them: https://wiki.php.net/rfc/pdo_escape_placeholders

I add a simple statement and add an example.

I am sorry the example is only valid using Postgresql database. But some question on multiple places (like [on stackoverflow](https://stackoverflow.com/q/36173440/1572236)) are related to this database (and I use often this RDBMS, I do not know if this is in use elsewhere).

### CODE REVIEW

File: reference/pdo/pdo/prepare.xml [hunk 1]
```diff
@@ -66,6 +66,9 @@
     which is natively supported by the driver.
    </simpara>
   </note>
+  <para>
+    Questions marks can be escaped by doubling them. That means that the “??” string will be translated to “?” when sending the query to the database.
```
Comment (kamil-tekiela) [inline reference/pdo/pdo/prepare.xml:70]:
```suggestion
    Question marks can be escaped by doubling them. That means that
    the <literal>??</literal> string will be translated to <literal>?</literal>
    when sending the query to the database.
```

File: reference/pdo/pdo/prepare.xml [hunk 2]
```diff
@@ -149,6 +152,22 @@ $red = $sth->fetchAll();
...
 ?>
+]]>
+    </programlisting>
+   </example>
+   <example>
+    <title>SQL statement template with question mark escaped</title>
+    <programlisting role="php">
+<![CDATA[
+<?php
+/* Execute a prepared statement which contains a question mark as sql statement */
```
Comment (kamil-tekiela) [inline reference/pdo/pdo/prepare.xml:163]:
```suggestion
/* Execute a prepared statement that contains a question mark */
```
Shouldn't that be the example description? Why is this comment necessary?

File: reference/pdo/pdo/prepare.xml [hunk 3]
```diff
@@ -149,6 +152,22 @@ $red = $sth->fetchAll();
...
+/* note: this is only valid on PostgreSQL databases */
+$sth = $dbh->prepare('SELECT '["a", "b", "c"]'::jsonb ?? ?);
+$sth->execute(array('a'));
+$resultTrue = $sth->fetchAll();
```
Comment (kamil-tekiela) [inline reference/pdo/pdo/prepare.xml:167]: The result is not used in the example. Why fetch the data at all?

File: reference/pdo/pdo/prepare.xml (hunk 2 above)
Comment (julienfastre) [reply reference/pdo/pdo/prepare.xml]: You are right, as we have a title, this is not necessary.

File: reference/pdo/pdo/prepare.xml (hunk 3 above)
Comment (julienfastre) [reply reference/pdo/pdo/prepare.xml]: I tryed to improve my example.

Thanks for your guidance.

**Comment by cmb69** [comment]
Thank you for the PR! I've applied some minor tweaks on top:
````.diff
 reference/pdo/pdo/prepare.xml | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/reference/pdo/pdo/prepare.xml b/reference/pdo/pdo/prepare.xml
index a23d05f822..ad1aa42d5c 100644
--- a/reference/pdo/pdo/prepare.xml
+++ b/reference/pdo/pdo/prepare.xml
@@ -67,7 +67,7 @@
    </simpara>
   </note>
   <para>
-    Question marks can be escaped by doubling them. That means that
+    As of PHP 7.4.0, question marks can be escaped by doubling them. That means that
     the <literal>??</literal> string will be translated to <literal>?</literal>
     when sending the query to the database.
   </para>
@@ -163,10 +163,10 @@ $yellow = $sth->fetchAll();
 <![CDATA[
 <?php
 /* note: this is only valid on PostgreSQL databases */
-$sth = $dbh->prepare('SELECT * FROM issues WHERE tag::jsonb ?? ?);
-$sth->execute(array('feature'));
+$sth = $dbh->prepare('SELECT * FROM issues WHERE tag::jsonb ?? ?');
+$sth->execute(['feature']);
 $featureIssues = $sth->fetchAll();
-$sth->execute(array('performance'));
+$sth->execute(['performance']);
 $performanceIssues = $sth->fetchAll();
 ?>
 ]]>
````

**Pull request closed without merging**